core/foundation/inc/RtypesCore.h

```diff
@@ -19,6 +19,17 @@
 //                                                                      //
 //////////////////////////////////////////////////////////////////////////
 
+/**
+ * \file
+ * \brief Basic types used by ROOT and required by TInterpreter.
+ * It ensures a portable fixed data type size across systems, since in
+ * the early days, sizeof(int) could be 2 or 4 depending on the architecture.
+ * \deprecated For future designs, unless for very specific needs, consider using instead standard
+ * fixed-width classes from <cstdint> such as `std::int16_t`, `std::int32_t` or from <cstdfloat> for floating types.
+ * \warning `Long_t` has not the same width across platforms, so it should be avoided if portability is envisioned.
```

Comment:
```suggestion
 * \warning `Long_t` has not the same width across platforms, so it should be avoided if portability is envisioned.
```
Technically one need to also avoid `std::int64_t` which on some platforms resolve to int and some platform resolve to long, so long different so ROOT I/O (and thus lead to a different StreamerInfo for the same class on the 2 platforms).  If I remember correctly `std::in32_t` always resolve to `int`, so might be fine.

So full fledge embrace of the std::type is actually waiting on ROOT I/O to be extended to support them explicitly.